backend/python/app/graphql/mutations/comment_mutation.py

```diff
@@ -17,3 +22,43 @@ def mutate(root, info, comment_data):
         comment_response = services["comment"].create_comment(comment=comment_data)
         ok = True
         return CreateComment(ok=ok, comment=comment_response)
+
+
+class UpdateCommentById(graphene.Mutation):
+    class Arguments:
+        comment_data = UpdateCommentDTO(required=True)
+
+    ok = graphene.Boolean()
+    comment = graphene.Field(lambda: CommentResponseDTO)
+
+    @require_authorization_by_role_gql({"User", "Admin"})
+    def mutate(root, info, comment_data):
+        try:
+            comment_response = services["comment"].update_comment(
+                updated_comment=comment_data
+            )
+            ok = True
+            return UpdateCommentById(ok=ok, comment=comment_response)
+        except Exception as e:
+            error_message = getattr(e, "message", None)
+            raise Exception(error_message if error_message else str(e))
+
+
+class UpdateComments(graphene.Mutation):
+    class Arguments:
+        comments_data = graphene.List(UpdateCommentDTO)
+
+    ok = graphene.Boolean()
+    comments = graphene.Field(lambda: graphene.List(UpdateCommentsResponseDTO))
```

Comment:
is `UpdateCommentsResponseDTO` necessary? Could we do a list of `CommentResponseDTO` since `UpdateCommentById` already returns that?

Reply:
follow-up to our discussion during work session today: after a search i believe graphql intentionally separates out input/output object types so i don't think i can use the same UpdateCommentDTO as the output
> The GraphQL Object type (ObjectTypeDefinition) defined above is inappropriate for re‐use here, because Object types can contain fields that define arguments or contain references to interfaces and unions, neither of which is appropriate for use as an input argument. For this reason, input objects have a separate type in the system.
http://spec.graphql.org/June2018/#sec-Input-Objects

i have kept the separate `UpdateCommentResponseDTO` for now until i find another way to do the update so we can return a list of `CommentResponseDTO`s